ImperatorToCK3/Mappers/Localizaton/LocalizationMapper.cs

```diff
@@ -0,0 +1,169 @@
+                return;
+            }
+            var fileNames = SystemUtils.GetAllFilesInFolderRecursive(languagePath);
+            foreach (var fileName in fileNames) {
+                var filePath = Path.Combine(languagePath, fileName);
+                try {
+                    var stream = File.OpenText(filePath);
+                    var reader = new BufferedReader(stream);
+                    ScrapeStream(reader, language);
+                    stream.Close();
+                } catch (Exception e) {
+                    Logger.Log(LogLevel.Warning, "Could not parse localization file " + filePath + " : " + e);
+                }
+            }
+        }
+        private void ScrapeStream(BufferedReader reader, string language) {
+            while (!reader.EndOfStream) {
+                var line = reader.ReadLine();
+                if (line == null || line.Length < 4 || line[0] == '#' || line[1] == '#') {
+                    continue;
+                }
+
+                var sepLoc = line.IndexOf(':');
+                if (sepLoc == -1) {
+                    continue;
+                }
+                var key = line.Substring(1, sepLoc - 1);
+                var newLine = line.Substring(sepLoc + 1);
+                var quoteLoc = newLine.IndexOf('\"');
+                var quote2Loc = newLine.LastIndexOf('\"');
+                if (quoteLoc == -1 || quote2Loc == -1 || quote2Loc - quoteLoc == 0) {
+                    continue;
+                }
+                var value = newLine.Substring(quoteLoc + 1, quote2Loc - quoteLoc - 1);
+
+                var gotValue = localizations.TryGetValue(key, out var locBlock);
+                if (gotValue) {
+                    if (language == "english")
+                        locBlock.english = value;
+                    if (language == "french")
+                        locBlock.french = value;
+                    if (language == "german")
+                        locBlock.german = value;
+                    if (language == "russian")
+                        locBlock.russian = value;
+                    if (language == "simp_chinese")
+                        locBlock.simp_chinese = value;
+                    if (language == "spanish")
+                        locBlock.spanish = value;
+                } else {
+                    var newBlock = new LocBlock();
+                    if (language == "english")
+                        newBlock.english = value;
+                    if (language == "french")
+                        newBlock.french = value;
+                    if (language == "german")
+                        newBlock.german = value;
+                    if (language == "russian")
+                        newBlock.russian = value;
+                    if (language == "simp_chinese")
+                        newBlock.simp_chinese = value;
+                    if (language == "spanish")
+                        newBlock.spanish = value;
```

Comment:
This is pretty much exactly like the code in the previous block. Can you put it in a function that both call?

Reply:
Done, see LocBlock.SetLocForLanguage.